src/order_cmd.cpp

```diff
@@ -824,6 +922,76 @@ void OrderList::MoveOrder(int from, int to)
 	this->ReindexOrderList();
 }
 
+std::string OrderList::ToJSONString()
+{
+
+	nlohmann::json json;
+	auto& SD_data = this->GetScheduledDispatchScheduleSet();
+
+	auto& headJson = json["head"];
+	for (unsigned int i = 0; auto &SD : SD_data) {
+
+		headJson["scheduled-dispatch"][i++] = nlohmann::json::parse(SD.ToJSONString());
+
+	}
+
+	const Order* o = this->GetFirstOrder();
+
+	if (o != nullptr) {
+		int i = 0;
+		do {
+			json["orders"][i++] = nlohmann::json::parse(o->ToJSONString());
+		} while ((o = this->GetNext(o)) != this->GetFirstOrder());
+	}
+	
+	return json.dump();
+
+}
+
+void OrderList::FromJSONString(const Vehicle * v,std::string json_str)
+{
+
+	Order errOrder;
+
+	errOrder.MakeLabel(OLST_TEXT);
+	errOrder.SetColour(COLOUR_RED);
+
+	nlohmann::json json;
+	try {
+		json = nlohmann::json::parse(json_str);
+	} catch(nlohmann::json::parse_error e){
+
+		ShowErrorMessage(STR_ERROR_JON, STR_ERROR_ORDERLIST_MALFORMED_JSON,WL_ERROR);
+		return;
+	}
+
+	//delete all orders before setting the new orders
+	DoCommandP(v->tile, v->index, v->GetNumOrders(), CMD_DELETE_ORDER | CMD_MSG(STR_ERROR_CAN_T_DELETE_THIS_ORDER));
```

Comment:
This breaks command estimation, and multiplayer, it's not valid to do this.

Reply:
The command scope should really be moved one level upwards because these commands are liable to create problems in multiplayer. Really importing should be atomic with respect to the command scope.